distributed/worker.py

```diff
@@ -3187,23 +3186,15 @@ def execute_task(task):
         return task
 
 
-try:
-    # a 10 MB cache of deserialized functions and their bytes
-    from zict import LRU
-
-    cache = LRU(10000000, dict(), weight=lambda k, v: len(v))
-except ImportError:
-    cache = dict()
+@functools.lru_cache(100)
+def cached_function_serialization(func):
+    return pickle.dumps(func)
 
 
 def dumps_function(func):
     """ Dump a function to bytes, cache functions """
     try:
-        result = cache[func]
-    except KeyError:
-        result = pickle.dumps(func)
-        if len(result) < 100000:
```

Comment:
I think that we still need to respect this limit and only cache small functions.  Some user functions get pretty pathological.

Reply:
Sure, I can see that being a nice feature to have. Do we want to include a similar condition on the function deserialization cache?

Reply:
Hrm, indeed.  I didn't see that we weren't doing that there.  It would be good to add, but given that we're not doing it in one place already if you wanted to skip and be done then that would seem reasonable.

Reply:
It's not possible to manually remove an item from the `functools.lru_cache` cache, so I added a small `LRU` class to `distributed.utils` which should avoid going into an inconsistent state if a `KeyboardInterrupt` is raised. 

Also, using an `LRU` `maxsize` of 100 + only caching small functions sets an upper bound on the size of the cache to be 10 MB (which is the current upper limit)